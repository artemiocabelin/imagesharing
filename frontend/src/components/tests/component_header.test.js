import React from 'react'
import { mount, shallow } from 'enzyme'
import Header from '../component_header'
import { StaticRouter } from 'react-router'

describe('Header', () => {
    let header;

    beforeEach(() => {
        header = shallow(<Header />)
    })

    it('renders properly', () => {
        expect(header).toMatchSnapshot()
    })

    it('contains a `title`', () => {
        expect(header.find('.title').text()).toEqual('ImageShare')
    })

    it('contains a menu`state`', () => {
        const initialMenu = []
        expect(header.state().menu).toEqual(initialMenu)
    })
    
    describe('when mounted', () => {
        const menuList = ['Home', 'Profile', 'Register', 'Login']
        beforeEach(() => {
            const context = {}
            header = mount(
                <StaticRouter context={context}>
                    <Header />
                </StaticRouter>
            )
        })

        it('contains menu links from menu `state`', () => {
            for(let i = 0; i < menuList.length; i++) {
                expect(header.find(`.${menuList[i]}`).exists()).toBe(true)
            }
        })
    })


})