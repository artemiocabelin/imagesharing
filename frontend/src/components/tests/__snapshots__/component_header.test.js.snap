// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          ImageShare
</h2>,
        <ul>
          
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ImageShare",
          "className": "title",
        },
        "ref": null,
        "rendered": "ImageShare",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            ImageShare
</h2>,
          <ul>
            
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ImageShare",
            "className": "title",
          },
          "ref": null,
          "rendered": "ImageShare",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
